### PR TITLE
overlord/state: implement State.Tasks() and Task.HaltTasks()

### PR DESCRIPTION
**Pull request by pedronis**
This implements State.Tasks() returning all known tasks and Task.HaltTasks() returning the tasks registered to wait on the receiver (symmetric to WaitTasks).

### CODE REVIEW

**Comment by zyga** [comment]
Looks good. +1

File: overlord/state/state.go [hunk 1]
```diff
@@ -238,6 +238,16 @@ func (s *State) Changes() []*Change {
...
 }
 
+// Tasks returns all tasks currently known to the state.
+func (s *State) Tasks() []*Task {
+	s.ensureLocked()
+	res := make([]*Task, 0, len(s.tasks))
+	for _, t := range s.tasks {
+		res = append(res, t)
+	}
+	return res
```
Comment (niemeyer) [inline overlord/state/state.go]: return append([]*Task(nil), s.tasks...)

File: overlord/state/state_test.go [hunk 2]
```diff
@@ -373,3 +385,36 @@ func (ss *stateSuite) TestEnsureBefore(c *C) {
...
+	st := state.New(nil)
+	st.Lock()
+	defer st.Unlock()
+
+	chg1 := st.NewChange("install", "...")
+	t11 := chg1.NewTask("check", "...")
+	t12 := chg1.NewTask("inst", "...")
+	chg2 := st.NewChange("remove", "...")
+	t21 := chg2.NewTask("check", "...")
+	t22 := chg2.NewTask("rm", "...")
```
Comment (niemeyer) [inline overlord/state/state_test.go]: Can we add t22 to chg1 too here? The expectation shouldn't change.

**Comment by niemeyer** [comment]
A couple of suggestions and LGTM

File: overlord/state/state.go (hunk 1 above)
Comment (pedronis) [inline overlord/state/state.go]: s.tasks is a map so I don't think it works

Comment (niemeyer) [inline overlord/state/state.go]: Duh, sorry.

**Pull request merged**